.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,72 @@
+name: 🔎 Unit tests
+on:
+  workflow_call:
+    inputs:
+      godot-version:
+        description: Godot Engine version to use for testing.
+        type: string
+        default: 4.2.1-stable
+      ss2d-ref:
+        description: A commit, branch or tag to use for testing.
+        type: string
+        required: true
+
+  workflow_dispatch:
+    inputs:
+      godot-version:
+        description: Godot Engine version to use for testing.
+        type: string
+        default: 4.2.1-stable
+      ss2d-ref:
+        description: A commit, branch or tag to use for testing.
+        type: string
+        required: true
+
+jobs:
+  unit-tests:
+    name: 🔎 Unit tests
+    runs-on: "ubuntu-latest"
+
+    env:
+      DONWLOAD_URL: https://github.com/godotengine/godot/releases/download/${{inputs.godot-version}}/Godot_v${{inputs.godot-version}}_linux.x86_64.zip
+      BIN: Godot_v${{inputs.godot-version}}_linux.x86_64
+
+    steps:
+      - name: Download Godot
```

Comment:
This step could be cached, no need to download and extract it on every run.

Reply:
Does it really matter, if you download and deploy a compressed cache, or a zip file from GH and unzip it? Not sure what difference would it make.

Reply:
Here's a link to the trial workflow runs I did some time ago. Check the time it takes to run that step - it's 2-3 seconds.
https://github.com/limbonaut/SmartShape2D/actions

Reply:
Saves time by not redownloading and uncompressing again.

Reply:
Ah, 2-3 seconds is fine I guess.